### PR TITLE
Make the current version work (again) with unsorted, non-aligned MDF4 files

### PR DESCRIPTION
**Pull request by kopytjuk**
Since >= 5.18.1 opening unsorted, non-byte aligned files was not possible. 

Especially https://github.com/danielhrisca/asammdf/blob/4aa9526246ed414cf96ef80644014c515f0fdb25/asammdf/blocks/mdf_v4.py#L9302 overwrote a variable `size` which was defined in a scope above which caused trouble.

In addition I (re-)added support for channel groups without `cg_flags` set. Why was it removed?

### CODE REVIEW

File: check_file.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import asammdf
```
Comment (danielhrisca) [inline check_file.py:1]: I don't think we need this file in the package

**Comment by danielhrisca** [comment]
> In addition I (re-)added support for channel groups without `cg_flags` set. Why was it removed?

I can't remember this change. Maybe it was accidental

**Comment by kopytjuk** [comment]
@danielhrisca Daniel, do you know what causes the problem in the Codacy/PR Quality Review check? I get errors in the "diff" and "deltas" check ...

**Comment by danielhrisca** [comment]
Thank you for the contribution Marat!

**Pull request merged**